src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -149,6 +150,42 @@
      */
     FeatureManager features;
 
+    /**
+     * Get the {@link ReferenceDataSource} for this {@link GATKTool}.
+     * Will throw a {@link GATKException} if the reference is null.
+     * @return the {@link ReferenceDataSource} for this {@link GATKTool}.  Never {@code null}.
+     */
+    protected ReferenceDataSource getReferenceDataSource() {
```

Comment:
After discussing with @lbergelson, it turns out that this change of introducing these datasource getters and then overriding them in the walker base classes to try to preserve encapsulation is slightly controversial and deserves separate discussion. Since this change is no longer actually needed for this PR (as far as I can determine), can I suggest that you:

1. In this branch, remove the new datasource getters from `GATKTool` and the overrides of the getters from the walker base classes.
2. Create a separate standalone PR that introduces the new getters, so that they can be reviewed/discussed independently of this `Funcotator` PR.
3. In the separate PR, instead of repeating the overrides of the datasource getters in every walker base class, could you introduce a new abstract `Walker` class that extends `GATKTool` and contains the overrides that throw an exception? Then you can just make the walker base classes like `ReadWalker` extend `Walker` rather than `GATKTool` directly, and you won't have to repeat the overrides all over the place. This change might improve the chances of the PR being accepted.

Reply:
Sounds good.  

Submitted as PR #4964